library/src/meta/document.rs

```diff
@@ -43,7 +45,10 @@ impl Construct for DocumentElem {
 
 impl LayoutRoot for DocumentElem {
     /// Layout the document into a sequence of frames, one per page.
+    #[tracing::instrument(name = "DocumentElem::layout_root", skip_all)]
     fn layout_root(&self, vt: &mut Vt, styles: StyleChain) -> SourceResult<Document> {
+        info!("Document layout");
```

Comment:
I think I'd prefer explicit `tracing::info!` everywhere instead of importing (same everywhere and for all levels).